implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/internal/EtsyStatsdLineBuilder.java

```diff
@@ -21,17 +21,17 @@
 import io.micrometer.core.instrument.config.NamingConvention;
 import io.micrometer.core.instrument.util.HierarchicalNameMapper;
 import io.micrometer.core.lang.Nullable;
-import org.pcollections.HashTreePMap;
-import org.pcollections.PMap;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 public class EtsyStatsdLineBuilder extends FlavorStatsdLineBuilder {
     private final HierarchicalNameMapper nameMapper;
-    private final Object namesLock = new Object();
     @SuppressWarnings({"NullableProblems", "unused"})
     private volatile NamingConvention namingConvention;
     @Nullable
     private volatile String nameNoStat;
-    private volatile PMap<Statistic, String> names = HashTreePMap.empty();
+    private final ConcurrentMap<Statistic, String> names = new ConcurrentHashMap<>();
```

Comment:
One side effect of this I believe is that iteration (which may happen in publishing) limits throughput on operations that are potentially generating new tags in another thread and adding meters to the map. I think...